cmake/wasm.cmake

```diff
@@ -0,0 +1,76 @@
+# wasm.cmake
+
+# Ensure we're using Emscripten
+if(NOT EMSCRIPTEN)
+    message(FATAL_ERROR "wasm.cmake should only be included when compiling with Emscripten")
+endif()
+
+# Set C++ standard for WASM build
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+# Emscripten compiler flags
+set(WASM_COMPILER_FLAGS
+    -s WASM=1
+    -s EXPORTED_RUNTIME_METHODS=['ccall','cwrap','FS','allocate']
+    -s ALLOW_MEMORY_GROWTH=1
+    -s EXPORTED_FUNCTIONS=['_malloc','_free']
+    -fexceptions
+    -s FORCE_FILESYSTEM=1
+)
+
+function(require_file FILE_PATH)
+    if(NOT EXISTS "${FILE_PATH}")
+        message(FATAL_ERROR "Required file not found: ${FILE_PATH}")
+    endif()
+endfunction()
+
+# Verify required files exist
+require_file("${CMAKE_CURRENT_SOURCE_DIR}/inputfile/elements.txt")
+require_file("${CMAKE_CURRENT_SOURCE_DIR}/inputfile/space_groups.txt")
+
+# Link flags specific to WASM
+set(WASM_LINK_FLAGS
+    -s ENVIRONMENT='web'
+    -s MODULARIZE=1
+    -s EXPORT_NAME='createMoloVolModule'
+    -s NO_EXIT_RUNTIME=1
+    -s ASSERTIONS=1
+    --bind
+    -s EXPORT_ES6=0#disabled for safari
+    -s USE_ES6_IMPORT_META=0
```

Comment:
I'm getting the following error when trying to build:
```console
em++: error: invalid command line setting `-sUSE_ES6_IMPORT_META=0`: Disabling is no longer supported
```
I'll enable the option and see whether that works.

Reply:
worked for me without the setting. Maybe it was required in the past but seems to be deprecated so I removed it.